### DISCUSSION TITLE
response()->streamDownload not working when called via event listener

### DISCUSSION THREAD

**Discussion by martinmildner** [livewire/livewire#3790] [7. Bugs (archived)]
Hi all,

After updating to Livewire 2.6.3 from 2.5.5 ```return response()->streamDownload...``` - when called via event listener -  is broken. I also tested it within a fresh Laravel installation:

```php
protected $listeners = ['download'];

public function download()
{
    return response()->streamDownload(function () {
        echo 'CSV Contents...';
    }, 'export.csv');
}
```

```php
$this->emit('download');
```

```download``` is called, but the download does not start.
(Browser: Safari Version 14.1.2) 

Does anyone else have this problem?

**Comment by joshhanley** [comment]
@martinmildner there was some issues with file downloads which has been fixed and tagged a new release 2.6.4.

Can you please test the latest version and let us know if it has fixed your issue or not?

Thanks!

**Reply by martinmildner** [reply]
Thanks @joshhanley, I tested it with 2.6.4, but unfortunately still does not work. It works with 2.5.5.

**Reply by joshhanley** [reply]
@martinmildner bugger, ok, thanks for checking.

If you look at the network tab of your browser dev tools, and then fire your event, can you see if you are getting a response from your component that has the download in it? 

If it does respond, can you look at the response by clicking on the request and then going to the "preview" tab and see if it is a file download response?

Thanks!

**Reply by martinmildner** [reply]
no, download ist not in den response.
2.5.5:
```php
{
    "effects": {
        "html": null,
        "download": {
            "name": "export.csv",
            "content": ......
```

2.6.5:
```php
{
    "effects": {
        "html": null,
        "dirty": []
    },
    "serverMemo": {
        "checksum": "10076c00dc03071bd041c3e5ef1beb7240808190747e7a45b47dd9d11bb9435f"
    }
}

**Reply by joshhanley** [reply]
@martinmildner thanks for testing! Yeah, ok something isn't right there. I will see if I can work out what is going on.

**Comment by joshhanley** [comment]
@martinmildner have submitted a failing test PR #3849 which demonstrates your issue.

Hope this helps!
